conda/core/solve.py

```diff
@@ -135,29 +138,34 @@ def solve_for_transaction(
             # is in the commented out get_install_transaction() function below. Exercised at
             # the integration level in the PrivateEnvIntegrationTests in test_create.py.
             raise NotImplementedError()
-        else:
-            unlink_precs, link_precs = self.solve_for_diff(
-                update_modifier,
-                deps_modifier,
-                prune,
-                ignore_pinned,
-                force_remove,
-                force_reinstall,
-                should_retry_solve,
-            )
-            stp = PrefixSetup(
+
+        unlink_precs, link_precs = self.solve_for_diff(
+            update_modifier,
+            deps_modifier,
+            prune,
+            ignore_pinned,
+            force_remove,
+            force_reinstall,
+            should_retry_solve,
+        )
+        # TODO: Only explicitly requested remove and update specs are being included in
+        #   History right now. Do we need to include other categories from the solve?
+
+        # plugins run any post-solve processes here before performing the transaction
+        for post_solve in context.plugin_manager.get_hook_results("post_solves"):
+            post_solve.action(self._repodata_fn, unlink_precs, link_precs)
```

Comment:
Let's move this into the plugin manager class as `invoke_post_solves` so we can reuse it elsewhere if we need.

```suggestion
        # plugins run any post-solve processes here before performing the transaction
        context.plugin_manager.invoke_post_solves(self._repodata_fn, unlink_precs, link_precs)
```

Untested method:

```python
    def invoke_post_solves(
            self,
            repodata_fn: str,
            unlink_precs: tuple[tuple[PackageRecord, ...]],
            link_precs: tuple[tuple[PackageRecord, ...]],
        ) -> None:
        """
        Invokes ``CondaPostSolve.action`` functions registered with ``conda_post_solves``.

        :param repodata_fn: name of the command that is currently being invoked
        :param unlink_precs: packages records to remove from an environment
        :param link_precs: packages records to add to an environment
        """
        for hook in self.get_hook_results("post_solves"):
            hook.action(repodata_fn, unlink_precs, link_precs)

```

Reply:
see https://github.com/conda/conda/pull/13053/commits/3c0f8b7a3de60bacf88580e25aa0da6e078f8834